charts/timescaledb-single/templates/statefulset-timescaledb.yaml

```diff
@@ -255,10 +255,10 @@ spec:
         {{- end }}
         envFrom:
         - secretRef:
-            name: {{ template "secrets_credentials" . }}
+            name: {{- if .Values.secrets.credentialsSecretName }} "{{ .Values.secrets.credentialsSecretName }}" {{- else }} {{ template "secrets_credentials" . }} {{- end}}
             optional: false
         - secretRef:
-            name: {{ template "secrets_pgbackrest" . }}
+            name: {{- if .Values.secrets.pgbackrestSecretName }} "{{ .Values.secrets.pgbackrestSecretName }}" {{- else }} {{ template "secrets_pgbackrest" . }} {{- end}}
```

Comment:
Couldn't you put this logic into the secrets_X templates in templates/_helpers.tpl? It'd keep it future-proof and make sure all uses of the secrets would pull from the configured XSecretName values then.

Reply:
ok it can be done